### PR TITLE
Allan Evans IC week 3

### PR DESCRIPTION
**Pull request by aevans27**
I got through Iterations 1,2,3,4.  I was in the zone and forgot to commit regularly but I'll make sure to do that on the next one.

### CODE REVIEW

File: lib/park.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  attr_reader :name,
+              :admission_price,
+              :vehicles,
+              :passengers,
+              :revenue
+  def initialize(name, admission_price)
+    @name = name
+    @admission_price = admission_price
+    @vehicles = []
+    @passengers = []
```
Comment (dcoleman21) [inline lib/park.rb:11]: You don't want `passengers` or `revenue` here. A park won't have either of those upon creation. They will come into play once a park has added a vehicle. So these we would want to build out as methods below that can iterate over the vehicles to get what we desire.

File: lib/park.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    @revenue = 0
+  end
+
+  def add_vehicle(vehicle)
+    @vehicles << vehicle
+  end
+
+  def get_vehicle_passengers
```
Comment (dcoleman21) [inline lib/park.rb:19]: See my comment above about having `@passengers` in your initialize. I would make this my `passengers` method that iterates over my vehicles to get to my passengers. 
```ruby
def passengers
    @vehicles.flat_map do |vehicle|
      vehicle.passengers
    end
 end
```

File: lib/park.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      @vehicles.each do |vehicle|
+        vehicle.passengers.each do |pass|
+          @passengers << pass
+        end
+      end
+      @passengers
+    end
+  end
+
+  def charge_adults
```
Comment (dcoleman21) [inline lib/park.rb:30]: See my comment about `@revenue` in your initialize method. I would make this my `revenue` method that can iterate over my passengers method. So something like:
```ruby
def revenue 
    adults = passengers.select do |passenger|
      passenger.adult?
    end
    adults.length * @admission_price 
end
```

File: lib/park.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        end
+      end
+      @revenue
+    else
+      @revenue
+    end
+  end
+
+  def all_attendees
+    names = []
```
Comment (dcoleman21) [inline lib/park.rb:44]: What other enumerable could you use?

File: lib/park.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    @passengers.each do |person|
+      names << person.name
+    end
+    names.sort
+  end
+
+  def minors
+    names = []
+    @passengers.each do |person|
+      if person.adult? == false
```
Comment (dcoleman21) [inline lib/park.rb:54]: You could get rid of the `if` by just calling 
```ruby
!passenger.adult?
```

File: lib/vehicle.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    @is_speeding = true
+  end
+
+  def add_passenger(passenger)
+    @passengers << passenger
+  end
+
+  def number_adults
+    adults = 0
+    @passengers.each do |pass|
```
Comment (dcoleman21) [inline lib/vehicle.rb:28]: Look at the `.count` enumerable.

File: spec/park_spec.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+    expect(state_park.name).to eq("State")
+    expect(state_park.admission_price).to eq(30)
+  end
+
+  it 'get vehicles' do
+    state_park = Park.new("State", 30)
+    expect(state_park).to be_an(Park)
+
+    expect(state_park.name).to eq("State")
```
Comment (dcoleman21) [inline spec/park_spec.rb:23]: You don't have to have these expects here.

File: spec/park_spec.rb [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    
+    expect(state_park.vehicles).to eq([])
+    state_park.add_vehicle(honda)
+    state_park.add_vehicle(mini)
+    expect(state_park.vehicles).to eq([honda, mini])
+
+    state_park.get_vehicle_passengers
+    expect(state_park.passengers).to eq([charlie, jude, taylor])
+
+    state_park.charge_adults
```
Comment (dcoleman21) [inline spec/park_spec.rb:64]: See my comments in your Park class

**Review by dcoleman21 — COMMENTED** [review]
Overall nice job. I left comments for you to review. Let me know if you have any questions.

File: lib/park.rb (hunk 1 above)
Comment (aevans27) [reply lib/park.rb:11]: Ok, I was a little confused about @revenue, in the instructions I thought it asked for a variable name that was an int. I initially built @revenue as a method, but re read the instructions and turned it into its own attribute.

File: lib/park.rb (hunk 4 above)
Comment (aevans27) [reply lib/park.rb:44]: .find_all l believe.

File: spec/park_spec.rb (hunk 7 above)
Comment (aevans27) [reply spec/park_spec.rb:23]: Ok, I'll change that.  I was under the impression to bring over not just initialized objects but also the tests as well.

**Pull request closed without merging**